keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ArtifactController.kt

```diff
@@ -38,31 +40,21 @@ class ArtifactController(
   fun submitArtifact(@RequestBody echoArtifactEvent: EchoArtifactEvent) {
     echoArtifactEvent.payload.artifacts.forEach { artifact ->
       if (artifact.type.equals(deb.toString(), true) && artifact.isFromArtifactEvent()) {
-        publisher.publishEvent(ArtifactEvent(listOf(artifact), emptyMap()))
+        debianArtifactSupplier.publishArtifact(ArtifactEvent(listOf(artifact), emptyMap()))
```

Comment:
Here's where `ArtifactController` delegates publishing of the events coming in from echo and igor to the corresponding `ArtifactSuppliers`.

Reply:
can you explain a bit more about this obligation? why do we need 2 different artifact publishers?

Reply:
So the way I originally implemented this was to simply have `ArtifactController` implement `ArtifactPublisher` and register both Debian and Docker as supported artifact types. That works just fine, but I suspected I would get criticism for adding more "business logic" into a controller, and so my compromise was to move out the implementation of the interface and delegate. Once I did that, it seemed better to me to create two separate classes than a single `DebianAndDockerArtifactPublisher` class, but now that I think about it, I could have just named it `BasicArtifactPublisher` or `BuiltinArtifactPublisher` or something like that...

Thoughts?

Reply:
How about `DefaultArtifactPublisher` ?  That's the general pattern we use elsewhere in Spinnaker when we want to provide a default implementation of an interface.  In this case it makes sense because debs and docker images are most likely always going to be supported by default in Keel (no extensibility required).